app/controllers/transfers_controller.rb

```diff
@@ -18,16 +18,19 @@ def index
   def create
     @transfer = current_organization.transfers.new(transfer_params)
 
-    if @transfer.valid?
+    if !@transfer.valid? || @transfer.from_id == @transfer.to_id
+      load_form_collections
+      @transfer.line_items.build if @transfer.line_items.empty?
+      flash[:error] = !@transfer.valid? ? "There was an error creating the transfer" : "'From' and 'To' Storage Locations must be different"
```

Comment:
This is a doing a lot of work to deal with 2 different situations.

I think we might be better suited to creating a custom validation that ensures the 2 ids arent the same, and then just checking for validity, like before.